### ISSUE TITLE
PImage.resize() not implemented in Processing 4??

### ISSUE THREAD

**Issue by jordirosa-p5** [benfry/processing4#200]
## Description

The resize method for the PImage class is not implemented (yet?). 

## Expected Behavior

PImage.resize() has always worked, resizing the image to the size provided. Code made using previous versions stops working, the documentation in the processing reference web shows resize() as a usable method, though.

## Current Behavior

Attempting to using the resize method on a PImage throws an exception, and looking at the Processing 4 code is easy to see why it's throwing it: 

  public void resize(int w, int h) {  // ignore
    throw new RuntimeException("resize() not implemented for this PImage type");
  }

## Your Environment

 

* Processing version: 4.0a3
* Operating System and OS version: macOS Big Sur
* Other information:

## Possible Causes / Solutions

Is this method going to be implemented in Processing 4?

**Comment by behreajj** [comment]
If you want to write one for yourself, it depends on what kind of algorithm you want to use (bicubic, bilinear, nearest neighbor). After reading this [Stack Overflow](https://stackoverflow.com/questions/17640173/implementation-of-bi-cubic-resize) discussion, I wrote this Processing [code](https://github.com/behreajj/CamZup/blob/8cc2f15a801f73df35c58591145d4dcdec171720/src/camzup/pfriendly/ZImage.java#L1269) for my own use.

Like the README says, functions that rely on AWT are being shifted around. `PImageAWT`, a sub-class of `PImage`, uses a resize function that relies on AWT.

https://github.com/processing/processing4/blob/cbdf55c96d6aedce4c4a36fec8bcd582caf73a7e/core/src/processing/awt/PImageAWT.java#L121

Whether this version of resize is called depends on which renderer you use and whether you use `loadImage` or `createImage`.

`PGraphics` [extend](https://github.com/processing/processing4/blob/cbdf55c96d6aedce4c4a36fec8bcd582caf73a7e/core/src/processing/core/PGraphics.java#L154) `PImage`. If a resize algorithm were implemented in `PImage`, all its sub-classes--various graphics renderers included--would need to be tested to see if the parent resize worked or needed an override. Check this issues section for topics on `surface.setResizable` and you'll see how the problems compound.

**Comment by benfry** [comment]
If you're using `createImage()` or `loadImage()`, you should never run into this. 

Can you share an example that causes this problem?

**Comment by behreajj** [comment]
```java
void setup() {
  size(256, 256, JAVA2D);
  PImage img = createImage(256, 256, ARGB);
  println(img.getClass());
  img.resize(256, 128);
}
```

**Comment by jordirosa-p5** [comment]
In my case, the image I need to resize is one I create by calling `g.get()`.

The visual effect i get from my app to create a nice blurry effect totally depends on the last lines in my `draw()` function. Here is the most simple version of it:
```

void setup() {
  fullScreen(P3D);
  frameRate(30);

  background(255,255,255);
  stroke(0);
}

void draw(){
  ellipse(width*0.5, height*0.5, 20, 20);
  
  PImage gBig = g.get();
  gBig.resize(width+2, height+2);
  set(-1, -1, gBig);
}
```

**Comment by benfry** [comment]
Perfect, thanks both. `resize()` should be working in both of those cases; I'll check into what's going on.

**Comment by jordirosa-p5** [comment]
Thank you!

**Comment by benfry** [comment]
Fixed for 4.0 alpha 4.
